bson/bsoncodec/default_value_encoders.go

```diff
@@ -309,12 +309,17 @@ func (dve DefaultValueEncoders) ArrayEncodeValue(ec EncodeContext, vw bsonrw.Val
 				return err
 			}
 
-			encoder, err := ec.LookupEncoder(reflect.TypeOf(e.Value))
+			ve := e.Value
+			if reflect.TypeOf(ve) == nil {
```

Comment:
Instead of going through the codec system, if we encounter `nil` let's just encode that directly to BSON null. Something like this:
```go
if e.Value == nil {
    err = vw.WriteNull()
    if err != nil {
        return err
    }
}
```

Reply:
Hi @skriptble ,

Done.

Also I've got a question, if you don't mind: how do you feel about extracting the same part of code from those two functions (encodeArray & encodeSlice) into separate unexported function? Something like this:

```
func encodeElement(ec EncodeContext, dw bsonrw.DocumentWriter, e primitive.E) error {
	vw, err := dw.WriteDocumentElement(e.Key)
	if err != nil {
		return err
	}

	if e.Value == nil {
		err = vw.WriteNull()
		if err != nil {
			return err
		}
		return nil
	}
	encoder, err := ec.LookupEncoder(reflect.TypeOf(e.Value))
	if err != nil {
		return err
	}

	err = encoder.EncodeValue(ec, vw, reflect.ValueOf(e.Value))
	if err != nil {
		return err
	}
	return nil
}
```

Reply:
That sounds good but let's condense this:
```go
if e.Value == nil {
	err = vw.WriteNull()
	if err != nil {
		return err
	}
	return nil
}
```
down to:
```go
if e.Value == nil {
	return vw.WriteNull()
}
```

Reply:
Yeah, sure :)

Reply:
Done